### PR TITLE
Use LinkGenerator instead of IUrlHelper in SignalMethodProvider

### PR DESCRIPTION
**Pull request by lampersky**
signalUrl() javascript helper method stopped working, I'm getting this error:
image

/cc @jtkech

### CODE REVIEW

**Review by jtkech — APPROVED** [review]

**Comment by jtkech** [comment]
Good catch, how could it work before ;)

File: src/OrchardCore.Modules/OrchardCore.Workflows/Http/Scripting/SignalMethodProvider.cs [hunk 1]
```diff
@@ -23,8 +25,9 @@ public SignalMethodProvider(WorkflowExecutionContext workflowContext, ISecurityT
                 {
                     var payload = !String.IsNullOrWhiteSpace(workflowContext.CorrelationId) ? SignalPayload.ForCorrelation(signal, workflowContext.CorrelationId) : SignalPayload.ForWorkflow(signal, workflowContext.WorkflowId);
                     var token = securityTokenService.CreateToken(payload, TimeSpan.FromDays(7));
-                    var urlHelper = serviceProvider.GetRequiredService<IUrlHelper>();
-                    return urlHelper.Action("Trigger", "HttpWorkflow", new { area = "OrchardCore.Workflows", token });
+                    var urlHelperFactory = serviceProvider.GetRequiredService<IUrlHelperFactory>();
```
Comment (ns8482e) [inline src/OrchardCore.Modules/OrchardCore.Workflows/Http/Scripting/SignalMethodProvider.cs:28]: How about using LinkGenerator?

Comment (jtkech) [reply src/OrchardCore.Modules/OrchardCore.Workflows/Http/Scripting/SignalMethodProvider.cs:28]: Yes good idea, which uses internally the urlHelper, as I remember.

Comment (lampersky) [reply src/OrchardCore.Modules/OrchardCore.Workflows/Http/Scripting/SignalMethodProvider.cs]: @ns8482e done

File: src/OrchardCore.Modules/OrchardCore.Workflows/Http/Scripting/SignalMethodProvider.cs [hunk 2]
```diff
@@ -23,8 +24,9 @@ public SignalMethodProvider(WorkflowExecutionContext workflowContext, ISecurityT
                 {
                     var payload = !String.IsNullOrWhiteSpace(workflowContext.CorrelationId) ? SignalPayload.ForCorrelation(signal, workflowContext.CorrelationId) : SignalPayload.ForWorkflow(signal, workflowContext.WorkflowId);
                     var token = securityTokenService.CreateToken(payload, TimeSpan.FromDays(7));
-                    var urlHelper = serviceProvider.GetRequiredService<IUrlHelper>();
-                    return urlHelper.Action("Trigger", "HttpWorkflow", new { area = "OrchardCore.Workflows", token });
+                    var httpContextAccessor = serviceProvider.GetRequiredService<IHttpContextAccessor>();
+                    var linkGenerator = serviceProvider.GetRequiredService<LinkGenerator>();
+                    return linkGenerator.GetPathByAction(httpContextAccessor.HttpContext, "Trigger", "HttpWorkflow", new { area = "OrchardCore.Workflows", token });
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Workflows/Http/Scripting/SignalMethodProvider.cs:29]: Please add an empty line before the return

**Review by hishamco — CHANGES_REQUESTED** [review]

**Review by ns8482e — APPROVED** [review]

**Review by hishamco — APPROVED** [review]

**Pull request merged**